ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/hateoas/HateoasMapperImpl.java

```diff
@@ -0,0 +1,188 @@
+ *
+ * @since 1.9.0
+ */
+public class HateoasMapperImpl implements HateoasMapper {
+    private final String packageId;
+    private final SemanticModel semanticModel;
+
+    public HateoasMapperImpl(String packageId, SemanticModel semanticModel) {
+        this.packageId = packageId;
+        this.semanticModel = semanticModel;
+    }
+
+    @Override
+    public void setSwaggerLinks(ServiceDeclarationNode serviceNode, OpenAPI openAPI) {
+        Optional<Symbol> serviceSymbolOpt = semanticModel.symbol(serviceNode);
+        if (serviceSymbolOpt.isEmpty()) {
+            return;
+        }
+        ServiceDeclarationSymbol serviceSymbol = (ServiceDeclarationSymbol) serviceSymbolOpt.get();
+        int serviceId = serviceSymbol.hashCode();
+        Paths paths = openAPI.getPaths();
+        for (Node node: serviceNode.members()) {
+            if (!node.kind().equals(SyntaxKind.RESOURCE_ACCESSOR_DEFINITION)) {
+                continue;
+            }
+            FunctionDefinitionNode resource = (FunctionDefinitionNode) node;
+            Optional<ApiResponses> responses = getApiResponsesForResource(resource, paths);
+            if (responses.isEmpty()) {
+                continue;
+            }
+            setSwaggerLinksInApiResponse(serviceId, resource, responses.get());
+        }
+    }
+
+    private Optional<ApiResponses> getApiResponsesForResource(FunctionDefinitionNode resource, Paths paths) {
+        String resourcePath = MapperCommonUtils.unescapeIdentifier(generateRelativePath(resource));
+        if (!paths.containsKey(resourcePath)) {
+            return Optional.empty();
+        }
+        PathItem openApiResource = paths.get(resourcePath);
+        String httpMethod = resource.functionName().toString().trim();
+        ApiResponses responses = null;
+        switch (httpMethod.trim().toUpperCase(Locale.ENGLISH)) {
+            case Constants.GET -> {
+                responses = openApiResource.getGet().getResponses();
+            }
+            case Constants.PUT -> {
+                responses = openApiResource.getPut().getResponses();
+            }
+            case Constants.POST -> {
+                responses = openApiResource.getPost().getResponses();
+            }
+            case Constants.DELETE -> {
+                responses = openApiResource.getDelete().getResponses();
+            }
+            case Constants.OPTIONS -> {
+                responses = openApiResource.getOptions().getResponses();
+            }
+            case Constants.PATCH -> {
+                responses = openApiResource.getPatch().getResponses();
+            }
+            case Constants.HEAD -> {
+                responses = openApiResource.getHead().getResponses();
+            }
+            default -> { }
+        }
+        return Optional.ofNullable(responses);
+    }
+
+    private void setSwaggerLinksInApiResponse(int serviceId, FunctionDefinitionNode resource,
```

Comment:
We have two types of links here, one is the hateoas links from the Ballerina side and the other is the links in the OpenAPI side. This word `swagger` is used to indicate its about OpenAPI links..

Reply:
As discussed with @TharmiganK instead of using `Swagger` changed it to `OpenApi` with this fix [#6324c6578e24d5ba975bd785c0c3c2b81a29be6a](https://github.com/ballerina-platform/openapi-tools/pull/1589/commits/6324c6578e24d5ba975bd785c0c3c2b81a29be6a)